mlflow/sagemaker/container/__init__.py

```diff
@@ -87,35 +99,52 @@ def _serve():
            "mlflow.sagemaker.container.scoring_server.wsgi:app").format(nworkers=cpu_count)
     bash_cmds.append(cmd)
     gunicorn = Popen(["/bin/bash", "-c", "; ".join(bash_cmds)])
-    signal.signal(signal.SIGTERM, lambda a, b: _sigterm_handler(nginx.pid, gunicorn.pid))
+    signal.signal(signal.SIGTERM, lambda a, b: _sigterm_handler(pids=[nginx.pid, gunicorn.pid]))
     # If either subprocess exits, so do we.
-    pids = set([nginx.pid, gunicorn.pid])
-    while True:
-        pid, _ = os.wait()
-        if pid in pids:
-            break
-    _sigterm_handler(nginx.pid, gunicorn.pid)
+    awaited_pids = _await_subprocess_exit_any(procs=[nginx, gunicorn])
+    _sigterm_handler(awaited_pids)
+
+
+def _serve_mleap(model):
+    mleap = Popen(["java", "-cp", "/opt/mlflow/mlflow/java/target/mlflow-java-"
```

Comment:
Maybe we should just see if a jar exists in a known location in the container? And build and move it into place if it did not already?

Reply:
I understand moving the jar into a known location. I'm not what the benefit would be of trying to build and move it once the container is already running; can you elaborate?